rudder-web/src/main/scala/com/normation/rudder/web/model/DirectiveFieldEditors.scala

```diff
@@ -754,29 +594,46 @@ class MasterPasswordField(val id: String, blankable:Boolean, algos:Seq[HashAlgoC
       s.updateValue(keepCurrentPwd, blankPwd, pastValue, newInput, chosenAlgo)
     }
 
-    currentValue = newInput
-
+    currentValue = Some(newInput)
     if(keepCurrentPwd) pastValue
     else if(blankPwd) ""
     else if(newInput == "") ""
     else chosenAlgo.serialize(newInput.getBytes())
   }
 
-
-  def parseClient(s: String): Unit = {
-    if (null == s) _x = "" else _x = s
+  def parseClient(s: String) = {
```

Comment:
could you put a comment there, to explain what is expected of this json format ?